src/backend/distributed/connection/placement_connection.c

```diff
@@ -592,14 +592,21 @@ FindPlacementListConnection(int flags, List *placementAccessList, const char *us
 								"modified over multiple connections")));
 			}
 		}
+		else if (accessType == PLACEMENT_ACCESS_SELECT &&
```

Comment:
I don't think so. Just to be sure. I ran the regression tests with a NOTICE in this code block and it only showed up when there were two reference table joins in the same query/transaction, most notably in the the CH-benCHmark test.

I don't think re-partition joins are special in terms of placement access, since we don't count things like data fetches as a placement access.